### PR TITLE
Improve logging of availability error to show retry interval

### PR DESCRIPTION
**Pull request by Martchus**
Based on #3640

---

It will look like this:
```
[warn] Worker cache not available: Cache service info error: Connection refused - checking again for web UI 'http://localhost:9526' in 100.00 s
```

See #3645

### CODE REVIEW

File: lib/OpenQA/Worker/WebUIConnection.pm [hunk 1]
```diff
@@ -420,19 +420,20 @@ sub send_status {
...
             my $status_update_interval = $self->_calculate_status_update_interval;
-            $self->{_send_status_timer} = Mojo::IOLoop->timer(
-                $status_update_interval,
-                sub {
-                    $self->send_status;
-                });
+            my $webui_host             = $self->webui_host;
+            log_warning "$status->{reason} - checking again for web UI '$webui_host' in $status_update_interval s"
+              if $status->{reason};
+            $self->{_send_status_timer} = Mojo::IOLoop->timer($status_update_interval, sub { $self->send_status });
```
Comment (kalikiana) [inline lib/OpenQA/Worker/WebUIConnection.pm:436]: Neat approach. And still simple!

How often does this re-try, though? Should it have a limit?

Comment (okurz) [reply lib/OpenQA/Worker/WebUIConnection.pm:436]: The actual retry behaviour is not impacted by this change. The retry happens on different intervals depending on the actual worker state and AFAIK it's without limit and so far is ok to not have a limit :)

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Comment by okurz** [comment]
hm, this now implicitly merged https://github.com/os-autoinst/openQA/pull/3640 now as well. We don't have full patch coverage.

@Martchus as long as your memory about your changes is still fresh can you please try to come up with 100% coverage for your recently modified code lines?

**Comment by Martchus** [comment]
Yes, I've seen it as well. Note that the code in `t/lib/OpenQA/Test/Utils.pm` is definitely run but can not be collected because it is a sub process. Not sure how gracefully we stop the sub process. Since it is just test code I tend to mark the block as uncoverable. Apart from that it is only the `$self->_setup_pool_directory` case in `check_availability`.

**Pull request merged**